monitoring/bin/change_grafana_admin_password.sh

```diff
@@ -0,0 +1,133 @@
+source monitoring/bin/common.sh
+
+this_script=`basename "$0"`
+
+function show_usage {
+  log_message  "Usage: $this_script [--namespace NAMESPACE --tenant TENANT]"
+  log_message  ""
+  log_message  "Changes the password of the Grafana admin user."
+  log_message  ""
+  log_message  "To change the Grafana admin user at the cluster level, do not pass any "
+  log_message  "arguments.  To change the Grafana admin user at the tenant level, you need"
+  log_message  "to provide the following arguments:"
+  log_message  "     -ns, --namespace NAMESPACE   - The namespace where the Viya tenant resides."
+  log_message  "     -t,  --tenant TENANT         - The tenant whose logging data source you want to set up."
+  log_message  "     -p,  --password PASSWORD         - The tenant whose logging data source you want to set up."
+  log_message  ""
+}
+
+# Assigning passed in parameters as variables for the script:
+POS_PARMS=""
+
+# Setting passed in variables:
+while (( "$#" )); do
+  case "$1" in
+    -ns|--namespace)
+      if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+        tenantNS=$2
+        shift 2
+      else
+        log_error "A value for parameter [NAMESPACE] has not been provided." >&2
+        show_usage
+        exit 2
+      fi
+      ;;
+    -t|--tenant)
+      if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+        tenant=$2
+        shift 2
+      else
+        log_error "A value for parameter [TENANT] has not been provided." >&2
+        show_usage
+        exit 2
+      fi
+      ;;
+    -p|--password)
+      if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+        password=$2
+        shift 2
+      else
+        log_error "A value for parameter [PASSWORD] has not been provided." >&2
+        show_usage
+        exit 2
+      fi
+      ;;
+    -h|--help)
+      show_usage
+      exit
+      ;;
+    -*|--*=) # unsupported flags
+      log_error "Unsupported flag $1" >&2
+      show_usage
+      exit 1
+      ;;
+    *) # preserve positional arguments
+      POS_PARMS="$POS_PARMS $1"
+      shift
+      ;;
+  esac
+done
+
+# Convert namespace and tenant to all lower-case
+tenantNS=$(echo "$tenantNS"| tr '[:upper:]' '[:lower:]')
+tenant=$(echo "$tenant"| tr '[:upper:]' '[:lower:]')
+
+# Check for parameters - set with cluster or tenant
+if [ -z "$tenantNS" ] && [ -z "$tenant" ]; then
+  cluster="true"
+  namespace=$MON_NS
+  v4mSecret="v4m-grafana"
+elif [ -n "$tenantNS" ] && [ -n "$tenant" ]; then
+  cluster="false"
+  namespace=$tenantNS
+  v4mSecret="v4m-grafana-${tenant}"
+else
+  log_error "Both a [NAMESPACE] and a [TENANT] are required in order to change the Grafana admin password.";
+  exit 1
+fi
+
+# Check and make sure that a password was provided.
+if [ -z $password ]; then
+  log_error "A value for parameter [PASSWORD] has not been provided"
+  exit 1
+fi 
+
+if [ "$cluster" == "true" ]; then
+  grafanaPod="$(kubectl get pods -n $MON_NS -l app.kubernetes.io/name=grafana --template='{{range .items}}{{.metadata.name}}{{end}}')"
```

Comment:
The reference to $MON_NS here is (technically) correct, but based on line 84 above, we probably should refer to $namespace here (since we took the time to set it).  It's not a big deal but if we need to change something later, this would force us to hunt for references to both.